shared/macros-ansible.jinja

```diff
@@ -0,0 +1,152 @@
+- name: "{{{ msg or rule_title }}}"
+  lineinfile:
+    path: '{{{ path }}}'
+    create: {{{ create }}}
+    {{%- if regex %}}
+    regexp: '{{{ regex }}}'
+    {{%- endif %}}
+    {{%- if state == 'present' %}}
+    line: '{{{ new_line }}}'
+    state: present
+    {{%- if insert_after %}}
+    insertafter: '{{{ insert_after }}}'
+    {{%- elif insert_before %}}
+    insertbefore: '{{{ insert_before }}}'
+    {{%- endif %}}
+    {{% else %}}
+    state: '{{{ state }}}'
+    {{%- endif %}}
+    {{%- if validate %}}
+    validate: '{{{ validate }}}'
+    {{%- endif %}}
+  {{%- if with_items %}}
+  with_items: '{{{ with_items }}}'
+  {{%- endif %}}
+  {{%- if register %}}
+  register: '{{{ register }}}'
+  {{%- endif %}}
+  {{%- if when %}}
+  when: '{{{ when }}}'
+  {{%- endif %}}
+{{%- endmacro %}}
+
+{{#
+  Check the file system status of an object. Not a full implementation.
+#}}
+{{%- macro ansible_stat(msg='', path='', register='') %}}
+- name: '{{{ msg or rule_title }}}'
+  stat:
+    path: '{{{ path }}}'
+  {{%- if register %}}
+  register: '{{{ register }}}'
+  {{%- endif %}}
+{{%- endmacro %}}
+
+{{#
+  Find files matching a particular value. Not a full implementation.
+#}}
+{{%- macro ansible_find(msg='', paths='', recurse='yes', follow='no', contains='', register='', when='') %}}
+- name: '{{{ msg or rule_title }}}'
+  find:
+    paths: '{{{ paths }}}'
+    recurse: '{{{ recurse }}}'
+    follow: '{{{ follow }}}'
+    {{%- if contains %}}
+    contains: '{{{ contains }}}'
+    {{%- endif %}}
+  {{%- if register %}}
+  register: '{{{ register }}}'
+  {{%- endif %}}
+  {{%- if when %}}
+  when: '{{{ when }}}'
+  {{%- endif %}}
+{{%- endmacro %}}
+
+{{#
+  A wrapper for adding one, unique line to a file. A regex must be specified
+  to tell if the line is unique. This is helpful in configuration files where
+  a single configuration parameter might have multiple values, but only one
+  value is approved. All lines matching the regex are first removed and then
+  the new line is appended to the file.
+#}}
+{{%- macro ansible_only_lineinfile(msg, path, line_regex, new_line, create='no', block=False, validate='', insert_after='', insert_before='') -%}}
+{{%- if block %}}
+- name: "{{{ msg or rule_title }}}"
+  block:
+    {{{ ansible_lineinfile("Deduplicate values from " + path, path, regex=line_regex, new_line=None, create='no', state='absent')|indent }}}
+    {{{ ansible_lineinfile("Insert correct line to " + path, path, regex=None, new_line=new_line, create=create, state='present', validate=validate, insert_after=insert_after, insert_before=insert_before)|indent }}}
+{{%- else %}}
+{{{ ansible_lineinfile("Deduplicate values from " + path, path, regex=line_regex, new_line=None, create='no', state='absent') }}}
+{{{ ansible_lineinfile("Insert correct line into " + path, path, regex=None, new_line=new_line, create=create, state='present', validate=validate) }}}
```

Comment:
Is there any reason for this second call of `ansible_lineinfile` not setting `insert_after`?